### ISSUE TITLE
RacerD should be configurable to list external known thread-safe classes

### ISSUE THREAD

**Issue by dsmiley** [facebook/infer#1732]
I have [a Java class that holds a public static final ConcurrentHashMap](https://github.com/apache/solr/blob/f08f7bb3ef90381078e427f0d164a9f13afe070c/solr/solrj/src/java/org/apache/solr/common/util/ObjectReleaseTracker.java#L46).  CHM is thread-safe but RacerD does not know this (I also read this was the case in a paper about RacerD).  RacerD [reports](https://github.com/apache/solr/pull/1341/files#r1103247308) a THREAD_SAFETY_VIOLATION due to mutations to this map.

I think the best way to remedy this is to support a configuration file for Infer/RacerD to allow me to designate classes as thread-safe.  I am assuming/hoping that RacerD has a concept of a class being thread-safe in the first place.

This approach could also be used as an _alternative_ to specifying `@ThreadSafe(enableChecks=false)` on source code within the project so that I needn't add Infer's custom annotations.

**Comment by ngorogiannis** [comment]
This should be possible to achieve with the `--racerd-ignore-classes` option or corresponding `.inferconfig` setting.

**Comment by ngorogiannis** [comment]
https://fbinfer.com/docs/next/man-infer-analyze#RACERD%20CHECKER%20OPTIONS

**Comment by dsmiley** [comment]
Thank you; this was helpful; I managed to ignore certain classes that are frequent culprits.  I'll leave it to you to close or not.  Ignoring classes is merely a work-around to the problem I posed.  The inability to convey that an external class is thread-safe  seems like a big/obvious problem to me; ConcurrentHashMap and other java.util.concurrent structures are used a lot in big software projects, in my experience.  There's also the `Collections.synchronizedMap(new HashMap())` wrapper.

Another work-around I have not tried is to use a 3rd party cache-like utility that does not implement Map.  I suspect RacerD has some built-in knowledge of what a Map is (it's in the JDK) but not something custom.
